### PR TITLE
Added ARM CI

### PR DESCRIPTION
**Pull request by justinboswell**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .builder/jdk.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                for paths in required_files:
+                    for path in paths:
+                        full_path = os.path.join(prefix, path)
+                        if glob.glob(full_path, recursive=True):
+                            found += 1
+                            break
+
+                if found >= len(required_files):
+                    print('Found existing JDK8 at {}'.format(prefix))
+                    env.variables['java_home'] = self.path = prefix
```
Comment (graebm) [inline .builder/jdk.py]: trivial: double-assignment is weird to me

File: .builder/jdk.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                    self.installed = True
+                    return
+
+        target = '{}-{}'.format(env.spec.target, env.spec.arch)
```
Comment (graebm) [inline .builder/jdk.py]: trivial: You meant to fall through here if we looked for, but didn't find, an existing local installation? In any case, add a comment where the code resumes explaining the circumstances

`# Install JDK. We didn't find (or we're not using) an existing installation.`

File: .builder/jdk.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        if target not in URLs:
+            print('WARNING: No pre-built binaries for {} are available'.format(target))
+            return
```
Comment (graebm) [inline .builder/jdk.py]: trivial: "...prebuillt JDK binaries..."

Also, why not raise an exception? Is this just how `Builder.Import.instal()` is supposed to work?

File: CMakeLists.txt [hunk 4]
```diff
@@ -111,10 +115,20 @@ file(GLOB CRT_JAVA_SRC
...
 set(TARGET_LIB_DIR "lib/${TARGET_OS}/${TARGET_ARCH}")
 
 # shared lib that contains the CRT and JNI bindings, to be loaded by java
 add_library(${PROJECT_NAME} SHARED ${CRT_JAVA_HEADERS} ${CRT_JAVA_SRC})
+aws_use_package(aws-c-http)
+aws_use_package(aws-c-mqtt)
+aws_use_package(aws-c-cal)
+aws_use_package(aws-c-auth)
+aws_use_package(aws-c-common)
+aws_use_package(aws-c-io)
```
Comment (graebm) [inline CMakeLists.txt]: What about aws-c-compression?

Comment (justinboswell) [reply CMakeLists.txt]: oh, this was an overreaction to not finding libcrypto, should revert

File: .builder/jdk.py (hunk 2 above)
Comment (justinboswell) [reply .builder/jdk.py]: There's a return. Basically, there's no way we CAN install the JDK, so I've made it say "please install JDK8 or greater and set JAVA_HOME"

File: .builder/jdk.py (hunk 3 above)
Comment (justinboswell) [reply .builder/jdk.py]: Done.

**Pull request merged**